packages/driver/src/cy/commands/navigation.js

```diff
@@ -297,6 +297,19 @@ const stabilityChanged = (Cypress, state, config, stable) => {
   state('onPageLoadErr', onPageLoadErr)
 
   const loading = () => {
+    const count = state('redirectionCount')
+    const limit = config('redirectionLimit')
+
+    if (count === limit) {
+      $errUtils.throwErrByPath('navigation.reached_redirection_limit', {
+        args: {
+          limit,
+        },
+      })
+    }
+
+    state('redirectionCount', count + 1)
```

Comment:
Doesn't this actually place a limit on the number of page loads of any type, not just redirections?

Since this doesn't reset, a user loading 25 pages in a test would go over the limit, right?

Reply:
You're right. If a user loaded page more than **20** times, it threw an error. At first, I thought it was not a bad idea, because we usually don't visit 20 pages in a single test. But now that I think about it, it's a bit cruel. 

So, I decided to change the behavior by "count the redirected `href`s and if a single URL is visited 20 times throw an error."

Reply:
> a bit cruel.

:laughing: yeah, precisely